src/Pages/components/Comment/ModifyComment.jsx

```diff
@@ -0,0 +1,80 @@
+import { getRequest, updateRequest } from "../../../API";
+import { useNavigate, useParams } from "react-router-dom";
+import { useEffect, useState } from "react";
+
+export default function ModifyComment() {
+  // state
+  const [comment, setComment] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [modified, setModified] = useState(false);
+
+  // navigation
+  const { commentId, postId } = useParams();
+  const navigate = useNavigate();
+
+  // on load: fetch comment
+  useEffect(() => {
+    const runEffect = async () => {
+      const { data, error } = await getRequest(`/post/${postId}/comment`);
+      if (error === null) {
+        setComment(data);
+      } else {
+        console.log(error);
+      }
+      setLoading(false);
+    };
+    runEffect();
+  }, [postId]);
+
+  async function submitCallback(payload) {
+    return await updateRequest(
+      `/post/${postId}/comment/${payload.id}`,
+      payload
+    ).then(({ data, error }) => {
+      if (data) {
+        navigate(`/post/${data.id}`);
+        setModified(true);
+        return null;
+      } else {
+        return error[0];
+      }
+    });
+  }
+
+  return (
+    <div>
+      <h1>Modify Your comment</h1>
+      {loading && <p>Loading...</p>}
+      {!loading && (
+        <form
+          onSubmit={async (e) => {
+            e.preventDefault();
+            const error = await submitCallback(comment);
+            if (error) {
+              console.log(error);
+            }
```

Comment:
It would have been better to abstract the logic here into a function, rather than have it inline